.github/workflows/publish.yml

```diff
@@ -1,6 +1,10 @@
 name: Publish Python Package
 
 on:
+  push:
+    branches: [main]
```

Comment:
Do we need the push trigger? From the if statements in the steps it seems like they just prevent the execution for push trigger

Reply:
It's enabled for test pypi. It will release a new version to test PyPI every time we push to main. Might be useful for debugs and tests. I saw it in some other repo and found it quite interesting.

Reply:
Oh. you use that for Test PyPI